### ISSUE TITLE
Document how the compiler version is inferred

### ISSUE THREAD

**Issue by paulrberg** [NomicFoundation/hardhat-vscode#135]
There's no explanation in the README for how this extension infers what compiler version to use. Does it do that using the Hardhat configuration file? Does it check the version of each Solidity file by parsing the pragma? If the former, can this extension handle monorepos as well, where multiple Hardhat configuration files are to be found?

Also, if the end-user would prefer to change the compiler version (for any reason whatsoever) can they do that at all?

Answering these questions in the README would be very helpful.

**Comment by kanej** [comment]
The compiler version used is based on the hardhat configuration file. In effect we run the hardhat compilation pipeline based on the hardhat config at the root of the workspace folder.

We don't have monorepo support currently (I have opened an issue here to track the feature: https://github.com/NomicFoundation/hardhat-vscode/issues/138). 

We run one language server per workspace folder, so a hack would be to open a separate vscode workspace folder for each package folder. However, when trying this in the [hifi](https://github.com/hifi-finance/hifi) monorepo I ran into this bug: #137.

Leaving both those issues aside, we will add more details in the readme to explain how we leverage hardhat config for things like the solidity version.
